src/bindings/python/src/openvino/frontend/jax/jaxpr_decoder.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# flake8: noqa
+# mypy: ignore-errors
+
+import jax.core
+from openvino.frontend.jax.py_jax_frontend import _FrontEndJaxDecoder as Decoder
+# from openvino.frontend.jax.py_jax_frontend import _Type as DecoderType
+from openvino.runtime import op, PartialShape, Type as OVType, OVAny, Shape
+from openvino.frontend.jax.utils import jax_to_ov_type_map
+
+import jax
+
+from typing import List
+import logging
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+
+class JaxprPythonDecoder (Decoder):
+    def __init__(self, jax_module, example_inputs=None, jax_eqn=None, name=None):
+        Decoder.__init__(self)
+        
+        self.jax_module = jax_module
+        self.example_inputs = example_inputs
+        self.name = name
+        if self.name is None:
+            self.name = "jax_module"
+        if self.example_inputs is None:
+            self.example_inputs = []
+        elif isinstance(self.example_inputs, tuple):
+            self.example_inputs = list(self.example_inputs)
+        elif not isinstance(self.example_inputs, list):
+            self.example_inputs = [self.example_inputs]
+        
+        # TODO: the `self.jaxpr` here is possible to be a jaxpr or a jax_eqn. Maybe we need a better design.
+        if jax_eqn is None:
+            self.jaxpr = jax.make_jaxpr(self.jax_module)(*self.example_inputs).jaxpr
```

Comment:
Please move the step `jax.make_jaxpr` to before the `convert_model`. And this way you won't need `jax_debug` function